Add missing YAML files to .gitattributes

FurkanKambay opened the pull request:
Fixes #847 

(The generated) .gitattributes file was missing some other YAML files besides `.prefab`, `.unity`, etc.

I created an empty project and created almost everything under `Assets`/`Create`. Here are the files: https://i.imgur.com/QODfnOr.png

While I'm not familiar with a lot of these extensions, Unity supports them, so they should be in .gitattributes.

Commit:
Add missing YAML files to .gitattributes

shana commented:
Cooooooooooooooool 😃 😃 😄 !

There are more (or so Unity peeps tell me), but the only way to know is to scrape the Unity source (or so Unity peeps tell me), so yay this is a great start!

shana commented:
![](https://cloud.githubusercontent.com/assets/310137/24752224/d9352bbc-1ac4-11e7-96cc-1a678bf0d5e7.gif)

FurkanKambay commented:
@shana
Hey! (IDK if I should post this on a closed PR but I didn't want to open an issue for this)

So, I kinda took a look at the Unity source code 😄  and found something.
I think this is kind of a central location for the file extensions:
[switch statement in `FindIconForFile()`](https://github.com/Unity-Technologies/UnityCsReference/blob/0cf7cdad8092152ba078b4e306b4182608de040d/Editor/Mono/InternalEditorUtility.cs#L25)

I extracted all the extensions in the `switch` statement and categorized them. I also added [the template in alexkaratarakis/gitattributes](https://github.com/alexkaratarakis/gitattributes/blob/1a244a639484cf55209aeede8d7fe6cec4f52ea9/Unity.gitattributes) to the equation.

Here's a JSON file that has the differences between all 3 repos: [gist](https://gist.github.com/FurkanKambay/228939374abd6a9f79c4486690cf3c0d)
(I recommend folding everything to see the structure)

Now, if we look at just the Unity-specific extensions that we don't have here, the list is short:
`asmdef, asmref, uss, uxml, prefs`.

And these ones in the template repo: `cubemap, unitypackage` (I think the other 2 aren't valid extensions)

Otherwise, there is quite a lot of difference...

In the Unity source code, there are more extensions for file types like images, video files, etc. We can ignore them or add some of them. (Not all of them are commonly used)

Also, there are extensions that we have but the Unity source code doesn't contain, which is concerning 😟

Anyway, do you think adding any of these extensions is worthwhile? If so, should I open a PR here or in the new repo?

Thanks!
